sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationSettingDeserializationHelper.java

```diff
@@ -181,30 +181,31 @@ private static FeatureFlagConfigurationSetting getFeatureFlagPropertyValue(JsonP
             if (token == JsonToken.VALUE_STRING) {
                 map.put(ID, parser.getText());
             }
+            token = parser.nextToken();
         }
         // description
-        token = parser.nextToken();
         if (token == JsonToken.FIELD_NAME && DESCRIPTION.equals(parser.getCurrentName())) {
             token = parser.nextToken();
             if (token == JsonToken.VALUE_STRING) {
                 map.put(DESCRIPTION, parser.getText());
             }
+            token = parser.nextToken();
         }
         // display name
-        token = parser.nextToken();
         if (token == JsonToken.FIELD_NAME && DISPLAY_NAME.equals(parser.getCurrentName())) {
```

Comment:
This field is optional. The value of token here is `enabled`. This resulted line 202 skipping the `enabled` and getting the `VALUE_TRUE`/`VALUE_FALSE`, so it is never set.